custom_components/rainpoint/coordinator.py

```
@@ -342,3 +248,185 @@ async def _async_update_data(self):
+        multiple_status: list | None = None
+        try:
+            multiple_status = await self._client.get_multiple_device_status(device_list)
+            _LOGGER.debug(
+                debug_with_version("multipleDeviceStatus successful, got data for %d devices"),
+                len(multiple_status) if multiple_status else 0,
+            )
+        except RainPointApiError:
+            # Surface API errors to the outer except RainPointApiError -> UpdateFailed wrapper
+            # so HA marks entities unavailable instead of silently falling back.
+            raise
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)